storage/block/result.go

```diff
@@ -111,6 +111,11 @@ func (s *fetchBlockMetadataResults) Reset() {
 }
 
 func (s *fetchBlockMetadataResults) Close() {
+	var zeroed FetchBlockMetadataResult
```

Comment:
We should do this inside `fetchBlockMetadataResultsPool.Put()` since it already calls `Reset` above and is consistent with the other changes where we zero out the underlying slice when putting back to the pool